### ISSUE TITLE
Lottie Crash EXC_BAD_ACCESS

### ISSUE THREAD

**Issue by mihaicris-adoreme** [airbnb/lottie-ios#1926]
## Which Version of Lottie are you using?
Lottie 4.1.0 (lottie-spm)

## Expected Behavior 
Animation to load 

## Actual Behavior
App is crashing

## Code
```swift
private func setupWheelAnimation() {
       // CRASH ON NEXT LINE        Thread 1: EXC_BAD_ACCESS (code=1, address=0x1e) 
        animationView.animation = LottieAnimation.named("wheel_animation")
        animationView.loopMode = .loop
        animationView.play()
    }
```

## Notes
Works as expected in Lottie 3.5.0

## Animation JSON

{"v":"5.5.1","fr":23.9759979248047,"ip":0,"op":191.999983381819,"w":563,"h":598,"nm":"deal wheel (1)","ddd":0,"assets":[{"id":"image_0","w":56,"h":56,"u":"images/","p":"img_0.png","e":0},{"id":"image_1","w":35,"h":52,"u":"images/","p":"img_1.png","e":0},{"id":"image_2","w":501,"h":502,"u":"images/","p":"img_2.png","e":0},{"id":"image_3","w":526,"h":526,"u":"images/","p":"img_3.png","e":0}],"layers":[{"ddd":0,"ind":1,"ty":2,"nm":"inner button","refId":"image_0","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[281,316,0],"ix":2},"a":{"a":0,"k":[28,28,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6}},"ao":0,"ip":0,"op":191.999983381819,"st":0,"bm":0},{"ddd":0,"ind":2,"ty":2,"nm":"DIAL","refId":"image_1","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":1,"k":[{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":11,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":13,"s":[-110]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":16,"s":[-45]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":18,"s":[-110]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":22,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":25,"s":[-110]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":29,"s":[10]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":31,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":32,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":35,"s":[-100]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":40,"s":[10]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":42,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":51,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":54,"s":[-90]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":56,"s":[-98]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":62,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":64,"s":[10]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":66,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":105,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":106,"s":[-110]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":108,"s":[-62]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":109,"s":[-110]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":111,"s":[-69]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":113,"s":[-110]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":116,"s":[-61]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":118,"s":[-110]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":123,"s":[-33]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":125,"s":[-110]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":132,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":134,"s":[-90]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":141,"s":[10]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":143,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":147,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":149,"s":[-90]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":155,"s":[10]},{"t":156.999986411175,"s":[0]}],"ix":10},"p":{"a":0,"k":[282.5,49,0],"ix":2},"a":{"a":0,"k":[17.5,26,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6}},"ao":0,"ip":0,"op":191.999983381819,"st":0,"bm":0},{"ddd":0,"ind":3,"ty":2,"nm":"spinner copy 5","refId":"image_2","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":1,"k":[{"i":{"x":[0.774],"y":[1.825]},"o":{"x":[0.613],"y":[0]},"t":0,"s":[0]},{"i":{"x":[0.14],"y":[1]},"o":{"x":[0.063],"y":[0.09]},"t":9,"s":[-14]},{"i":{"x":[0.833],"y":[1]},"o":{"x":[0.167],"y":[0]},"t":83,"s":[279]},{"i":{"x":[0.703],"y":[1.667]},"o":{"x":[0.683],"y":[0]},"t":94,"s":[279]},{"i":{"x":[0.074],"y":[1]},"o":{"x":[0.072],"y":[0.052]},"t":102,"s":[265]},{"t":185.999983901137,"s":[720]}],"ix":10},"p":{"a":0,"k":[281.5,313,0],"ix":2},"a":{"a":0,"k":[250.5,251,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6}},"ao":0,"ip":0,"op":191.999983381819,"st":0,"bm":0},{"ddd":0,"ind":4,"ty":2,"nm":"purple outer circle","refId":"image_3","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[281,315,0],"ix":2},"a":{"a":0,"k":[263,263,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6}},"ao":0,"ip":0,"op":191.999983381819,"st":0,"bm":0}],"markers":[]}

**Comment by calda** [comment]
Can you share a sample project that reproduces this issue?

**Comment by calda** [comment]
Ah, I can reproduce this when running an app using lottie-spm on a physical device. Thanks for the report.

**Comment by calda** [comment]
Here's a fix: https://github.com/airbnb/lottie-spm/pull/5. We'll release 4.1.1 with this fix later today.

**Comment by mihaicris-adoreme** [comment]
Thanks, @calda !

**Comment by mihaicris-adoreme** [comment]
Tested version 4.1.1, still happening.  :(

```sh
Thread 1 Queue : com.apple.main-thread (serial)
#0	0x00000001c2a871b4 in objc_opt_class ()
#1	0x00000001c39c2d54 in swift_getObjectType ()
#2	0x00000001137ff274 in ___lldb_unnamed_symbol8467 ()
#3	0x0000000100b23340 in NotificationsOnboardingViewController.setupWheelAnimation() 
#4	0x0000000100b23120 in NotificationsOnboardingViewController.viewDidLoad()
```

**Comment by calda** [comment]
Can you share a copy of a project that reproduces that issue? Are you embedding the Lottie framework in your app target? 

Like this:

<img width="640" alt="image" src="https://user-images.githubusercontent.com/1811727/213792055-cfb40fdc-cada-46bf-9917-65aed4d9eaed.png">

**Comment by mihaicris-adoreme** [comment]
Unfortunately I can't share the project.

**Comment by mihaicris-adoreme** [comment]
<img width="710" alt="image" src="https://user-images.githubusercontent.com/68323356/213792467-2357adb2-e2a5-4841-909a-5d0b4b8bc80b.png">

**Comment by mihaicris-adoreme** [comment]
<img width="844" alt="image" src="https://user-images.githubusercontent.com/68323356/213792588-15f0d11d-1ae6-43c9-a278-7e59c2669c90.png">

Content of App application bundle

**Comment by mihaicris-adoreme** [comment]
<img width="636" alt="image" src="https://user-images.githubusercontent.com/68323356/213792792-d4b72e18-4e68-4a43-872a-a67402f05952.png">

Package dependency in Xcode

**Comment by mihaicris-adoreme** [comment]
<img width="1130" alt="image" src="https://user-images.githubusercontent.com/68323356/213794347-1731cf07-d6f5-4ada-a5bf-6e71d5eff593.png">

Buiding with RELEASE mode reveals this error.

**Comment by calda** [comment]
I'm not able to reproduce this problem with the Example app in the lottie-spm repo, including when building to a physical device with a release configuration. Could you create a standalone sample Xcode project that reproduces this issue for you and share that sample project?

**Comment by honkmaster** [comment]
Maybe the crash of @mihaicris-adoreme is related to my reported issue. 
https://github.com/airbnb/lottie-spm/pull/5#issuecomment-1398991609

**Comment by mihaicris-adoreme** [comment]
Tested with non-SPM repo and 4.1.1 works fine. - I'll skip the SPM version for the moment.

I was not able to reproduce with a blank project, although I tried, no crash, however the animation didn't even start. LottieAnimationView is there as a subview, but transparent. No errors in the console.

Maybe other users will reproduce and send a project sample.

**Comment by mihaicris-adoreme** [comment]
I am using Tuist to generate the project, maybe could be a reason, however, I have many other xcframework dependencies and no issues. (e.g. Firebase, etc)

**Comment by SajjadKharrazi** [comment]
i added lottie-spm repository with 4.1.1 version.
i can run app in physical and simulator device.
but i cant archive my app and i get this error:

<img width="349" alt="Screenshot 2023-01-21 at 19 20 45" src="https://user-images.githubusercontent.com/7239983/213875100-3d994d76-1d0c-4909-8fe3-9cbbd66539b4.png">

**Comment by calda** [comment]
@SajjadKharrazi, can you test the potential fix I mentioned here: https://github.com/airbnb/lottie-spm/pull/5#issuecomment-1399026263? According to @honkmaster that change fixes a similar problem he was encountering.

**Comment by SajjadKharrazi** [comment]
yes it works

**Comment by calda** [comment]
Great! Will make this change and then probably release 4.1.2 on Monday

**Comment by Jamonek** [comment]
Thanks! This was driving me crazy haha

**Comment by calda** [comment]
Published [lottie-spm 4.1.2](https://github.com/airbnb/lottie-spm/releases/tag/4.1.2) with a fix for this issue. Thanks for the report and working with us to find a solution!

**Comment by mihaicris-adoreme** [comment]
I confirm my issue is resolved with 4.1.2 Thanks!
